src/v/iceberg/schema.h

```diff
@@ -0,0 +1,27 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "container/chunked_hash_map.h"
+#include "iceberg/datatypes.h"
+#include "utils/named_type.h"
+
+#include <seastar/core/sstring.hh>
+
+namespace iceberg {
+
+struct schema {
+    using id_t = named_type<int32_t, struct schema_id_tag>;
+    struct_type sstruct;
```

Comment:
Please avoid abbreviations in the public interface unless they're really obvious. Is this `schema_struct`?

Reply:
Will do